### PR TITLE
Replace all Rlp usages with UntrustedRlp except for ethcore views

### PR DESCRIPTION
**Pull request by ascjones**
Part of #8033, replacing Rlp with UntrustedRlp.

Now everywhere (except ethcore views) are using `UntrustedRlp` and handling the `Result<_,DecoderError>`. 

This paves the way for the next phase, which would be to replace Rlp with UntrustedRlp in the ethcore views. Once that is done we can remove the legacy Rlp implementation.

### CODE REVIEW

File: ethcore/light/src/client/header_chain.rs [hunk 1]
```diff
@@ -74,6 +74,38 @@ pub struct BlockDescriptor {
 	pub total_difficulty: U256,
 }
 
+// best block data
+struct BestAndLatest {
```
Comment (debris) [inline ethcore/light/src/client/header_chain.rs]: handwritten implementations of `Encodable` and `Decodable` can be replaced with 

```
#[derive(RlpEncodable, RlpDecodable)]
struct {
}
```

File: ethcore/src/client/test_client.rs [hunk 2]
```diff
@@ -469,7 +469,7 @@ impl ChainInfo for TestBlockChainClient {
 impl BlockInfo for TestBlockChainClient {
 	fn block_header(&self, id: BlockId) -> Option<encoded::Header> {
 		self.block_hash(id)
-			.and_then(|hash| self.blocks.read().get(&hash).map(|r| Rlp::new(r).at(0).as_raw().to_vec()))
+			.and_then(|hash| self.blocks.read().get(&hash).map(|r| UntrustedRlp::new(r).at(0).unwrap().as_raw().to_vec()))
```
Comment (tomusdrw) [inline ethcore/src/client/test_client.rs]: We fought hard to get rid of `unwraps()` in the code base and this PR intorudces it back, can we please replace them with `expects()`?

File: ethcore/light/src/client/header_chain.rs (hunk 1 above)
Comment (ascjones) [reply ethcore/light/src/client/header_chain.rs]: Of course! Fixed.

File: ethcore/src/client/test_client.rs (hunk 2 above)
Comment (ascjones) [reply ethcore/src/client/test_client.rs]: Yes sorry that unwrap snuck in. Removed.

Comment (tomusdrw) [reply ethcore/src/client/test_client.rs]: There seems to be way more than that one. It's fine in tests code, but would avoid it in regular code.

Comment (ascjones) [reply ethcore/src/client/test_client.rs]: I've removed the unwraps I added in this PR. There are plenty in there still from before - like you say probably okay for tests. Which is why I was lazy in this case and just added them. Broken windows and all!

File: sync/src/blocks.rs [hunk 3]
```diff
@@ -293,9 +294,9 @@ impl BlockCollection {
 				let mut block_rlp = RlpStream::new_list(3);
 				block_rlp.append_raw(&block.header, 1);
```
Comment (tomusdrw) [inline sync/src/blocks.rs]: `new_from_header_and_body()`?

Comment (ascjones) [reply sync/src/blocks.rs]: Good idea. See latest commit.

**Pull request merged**